cpp/src/arrow/compute/api_scalar.h

```diff
@@ -218,11 +218,9 @@ enum CompareOperator : int8_t {
   LESS_EQUAL,
 };
 
-class ARROW_EXPORT CompareOptions : public FunctionOptions {
```

Comment:
Perhaps we should deprecate the struct and CompareOperator enum as well.

Reply:
On second thought, deprecating the enum would make some of the tests/benchmarks below annoying to write and it would be a tedious refactor. So let's not do that here - since this is about GLib - once this removes the GLib dependency we can go in and fully clean up CompareOptions.

Reply:
OK. I'll merge this as-is.
Let's do extra work as a follow-up task.

Reply:
Thank you! I filed ARROW-13378.